cadence/activity_loop.py

```diff
@@ -27,69 +28,85 @@ def activity_task_loop(worker):
+            polling_total = polling_end - polling_start
+            logger.debug(
+                f"PollForActivityTask: done. Start: {polling_start.total_seconds()}, End: {polling_end.total_seconds()}, Total: {polling_total.total_seconds()}"
+            )
+
             # Process activity task
-            # TODO: Add processing polling time
-            try:  
-                ret = handle_task(task, worker.activities)
-                respond = RespondActivityTaskCompletedRequest(
+            try:
+                ret = handle_task(task,worker.activities)
+                response = RespondActivityTaskCompletedRequest(
                     task_token=task.task_token,
-                    result = json.dumps(ret),
-                    identity = WorkflowService.get_identity(),
+                    result=json.dumps(ret),
+                    identity=WorkflowService.get_identity(),
                 )
-                _, err = service.respond_activity_task_completed(respond)
-                if err:
-                    logger.error("Error invoking RespondActivityTaskCompleted: %s", error)
+                service.respond_activity_task_completed(response)
             except Exception as ex:
-                logger.error(f"Activity {task.activity_type.name} failed: {type(ex).__name__}({ex})",exc_info=1)
-                respond: RespondActivityTaskFailedRequest 
-                respond = RespondActivityTaskFailedRequest(
+                logger.error(
+                    f"Activity {task.activity_type.name} failed: {type(ex).__name__}({ex})",
+                    exc_info=1)
+                response: RespondActivityTaskFailedRequest
+                response = RespondActivityTaskFailedRequest(
                     task_token=task.task_token,
                     reason="SOMTHING WENT WRONG",
                     identity=WorkflowService.get_identity(),
                     details=json.dumps({
-                        "detailMessage": f"Python error: {type(ex).__name__}({ex})"
-                    })
-                )
-                _, error = service.respond_activity_task_failed(respond)
-                if error:
-                    logger.error("Error invoking RespondActivityTaskFailed: %s", error)
-                    
+                        "detailMessage":
+                        f"Python error: {type(ex).__name__}({ex})"
+                    }))
+                try:
+                    service.respond_activity_task_failed(response)
+                except Exception as ex:
+                    logger.error("Error invoking RespondActivityTaskFailed: %s", ex)
+                    continue
+
             logger.info("Process ActivityTask: done")
-            
+    except Exception as ex:
+        logger.fatal(f"activity_task_loop: Uncought exception in main loop.",exec_info=True)
     finally:
         worker.notify_thread_stopped()
 
 
-def handle_task(task, activities):
+
+
+
+def handle_task(task, activities) -> Dict:
```

Comment:
Can we set types for args here for linting?

Reply:
yes